src/python/pants/invalidation/cache_manager.py

```diff
@@ -168,42 +184,46 @@ def _results_dir_path(self, root_dir, key, stable):
         self._STABLE_DIR_NAME if stable else sha1(key.hash).hexdigest()[:12]
     )
 
-  def create_results_dir(self, root_dir, allow_incremental):
-    """Ensures that a results_dir exists under the given root_dir for this versioned target.
+  def create_results_dir(self, root_dir):
+    """Ensures that a cleaned results_dir exists for invalid versioned targets.
 
-    If incremental=True, attempts to clone the results_dir for the previous version of this target
-    to the new results dir. Otherwise, simply ensures that the results dir exists.
+    Only guarantees results_dirs for invalid VTs.
     """
-    # Generate unique and stable directory paths for this cache key.
-    current_dir = self._results_dir_path(root_dir, self.cache_key, stable=False)
-    self._current_results_dir = current_dir
-    stable_dir = self._results_dir_path(root_dir, self.cache_key, stable=True)
-    self._results_dir = stable_dir
-    if self.valid:
-      # If the target is valid, both directories can be assumed to exist.
-      return
-
-    # Clone from the previous results_dir if incremental, or initialize.
-    previous_dir = self._use_previous_dir(allow_incremental, root_dir, current_dir)
-    if previous_dir is not None:
-      self.is_incremental = True
-      self._previous_results_dir = previous_dir
-      shutil.copytree(previous_dir, current_dir)
-    else:
-      safe_mkdir(current_dir)
+    self._current_results_dir = self._results_dir_path(root_dir, self.cache_key, stable=False)
+    self._results_dir = self._results_dir_path(root_dir, self.cache_key, stable=True)
 
-    # Finally, create the stable symlink.
-    relative_symlink(current_dir, stable_dir)
+    # If the target is valid, both directories are assumed to exist.
```

Comment:
I agree - I have a followup with that change. My hope was to change the behavior here as little as possible in the change, since the coverage was light and we were scheduling to backport to a stable branch.

But I ended up needing to make somewhat significant changes due to moving the previous check, so I can go ahead and add that here safely, I think.